File: lib/theme/typography.dart
import 'package:flutter/material.dart';
import 'tokens.dart';

abstract class AppTypography {
  static const String fontFamily = 'sans-serif';

  static const TextStyle displayLarge = TextStyle(
    fontSize: 32.0,
    fontWeight: FontWeight.bold,
    height: 1.2,
    color: AppColors.ink,
    letterSpacing: -0.5,
  );

  static const TextStyle displayMedium = TextStyle(
    fontSize: 28.0,
    fontWeight: FontWeight.bold,
    height: 1.25,
    color: AppColors.ink,
    letterSpacing: -0.25,
  );

  static const TextStyle headingLarge = TextStyle(
    fontSize: 24.0,
    fontWeight: FontWeight.w700,
    height: 1.3,
    color: AppColors.ink,
  );

  static const TextStyle headingMedium = TextStyle(
    fontSize: 20.0,
    fontWeight: FontWeight.w600,
    height: 1.35,
    color: AppColors.ink,
  );

  static const TextStyle bodyLarge = TextStyle(
    fontSize: 16.0,
    fontWeight: FontWeight.w400,
    height: 1.5,
    color: AppColors.ink,
  );

  static const TextStyle bodyMedium = TextStyle(
    fontSize: 14.0,
    fontWeight: FontWeight.w400,
    height: 1.45,
    color: AppColors.ink,
  );

  static const TextStyle caption = TextStyle(
    fontSize: 12.0,
    fontWeight: FontWeight.w400,
    height: 1.4,
    color: AppColors.ink,
  );

  static TextTheme toTextTheme() {
    return const TextTheme(
      displayLarge: displayLarge,
      displayMedium: displayMedium,
      headlineLarge: headingLarge,
      headlineMedium: headingMedium,
      bodyLarge: bodyLarge,
      bodyMedium: bodyMedium,
      bodySmall: caption,
      labelLarge: TextStyle(
        fontSize: 16.0,
        fontWeight: FontWeight.w600,
        color: AppColors.ink,
      ),
      labelMedium: TextStyle(
        fontSize: 14.0,
        fontWeight: FontWeight.w600,
        color: AppColors.ink,
      ),
      labelSmall: TextStyle(
        fontSize: 12.0,
        fontWeight: FontWeight.w600,
        color: AppColors.ink,
      ),
    );
  }
}

File: src/theme/typography.ts
// src/theme/typography.ts
import { colors } from './tokens';

export const typography = {
  fontFamily: {
    serif: 'System', // Display serif placeholder
    sans: 'System',  // Body sans placeholder
  },
  styles: {
    displayLarge: {
      fontSize: 32,
      lineHeight: 40,
      fontWeight: 'bold' as const,
      color: colors.ink,
    },
    displayMedium: {
      fontSize: 28,
      lineHeight: 36,
      fontWeight: 'bold' as const,
      color: colors.ink,
    },
    headingLarge: {
      fontSize: 24,
      lineHeight: 32,
      fontWeight: '700' as const,
      color: colors.ink,
    },
    headingMedium: {
      fontSize: 20,
      lineHeight: 28,
      fontWeight: '600' as const,
      color: colors.ink,
    },
    bodyLarge: {
      fontSize: 16,
      lineHeight: 24,
      fontWeight: '400' as const,
      color: colors.ink,
    },
    bodyMedium: {
      fontSize: 14,
      lineHeight: 20,
      fontWeight: '400' as const,
      color: colors.ink,
    },
    caption: {
      fontSize: 12,
      lineHeight: 18,
      fontWeight: '400' as const,
      color: colors.ink,
    },
  },
} as const;
